src/components/projects/ProjectTasks/NewTask/index.tsx

```diff
@@ -0,0 +1,359 @@
+'use client';
+
+// CSS imports
+import './NewTask.css';
+import { ITasks, Status } from '@/types/database.interface';
+import { useContext, useEffect, useState } from 'react';
+import { TaskContext } from '../ContextProvider';
+import { TaskMemberInput } from '../client';
+import { create } from 'domain';
+import {
+	createTask,
+	getProjectMembersTasks,
+	getAllTaskMembers,
+	addTaskMember,
+	removeMemberFromTask,
+} from '@/lib/actions/client';
+
+export const EditTaskForm = ({
+	role,
+	setRefresh,
+	newTask = false,
+	project_id,
+	org_id,
+	dismiss,
+}: {
+	role?: string;
+	setRefresh: any;
+	newTask?: boolean;
+	project_id: any;
+	org_id: any;
+	dismiss: any;
+}) => {
+	// Form state
+	// Holds the user object, client action lib removeTaskMember task id, to remove the the task_member user_id, only admin/adminstrators can change/edit task members makes span task
+	// Create an opt task field
+	//1. create optional field in editTaskForm (role)
+	//2. Pass that to member badge input
+	//3. Use prop here inside of badge and display X on hover that has functionality
+
+	const contxt = useContext(TaskContext);
+
+	const {
+		tasks,
+		setTasks,
+		addTask,
+		removeTask,
+		updateTask,
+		selectedTask,
+		setSelectedTask,
+	} = !newTask
+		? contxt
+		: {
+				tasks: '',
+				setTasks: '',
+				addTask: '',
+				removeTask: '',
+				updateTask: contxt.updateTask,
+				selectedTask: '',
+				setSelectedTask: contxt.setSelectedTask,
+		  };
+
+	const [teamMember, setTeamMember]: any = useState();
+
+	const [taskMembers, setTaskMembers]: any = useState([]);
+
+	const [members, setMembers] = useState([
```

Comment:
Remove this if not using the new portion here the dropdown selection of members.

Reply:
so this section is not used for drop down selection of members but it is used in other areas of the code